File: architecturecomponents/src/main/java/com/fesskiev/architecturecomponents/di/AppComponent.java
package com.fesskiev.architecturecomponents.di;

import com.fesskiev.architecturecomponents.domain.source.location.LocationService;
import com.fesskiev.architecturecomponents.ui.cities.CitiesViewModel;
import com.fesskiev.architecturecomponents.ui.map.MapViewModel;
import com.fesskiev.architecturecomponents.ui.weather.WeatherViewModel;

import javax.inject.Singleton;

import dagger.Component;

@Component(modules = {
        AppModule.class,
        RxBusModule.class,

        RepositoryModule.class,
        DataSourceModule.class,
        NetworkModule.class,
        DatabaseModule.class})

@Singleton
public interface AppComponent {
    void inject(CitiesViewModel viewModel);
    void inject(WeatherViewModel viewModel);
    void inject(MapViewModel viewModel);
    void inject(LocationService service);
}

File: graphql/src/main/kotlin/com/fesskiev/graphql/di/AppComponent.kt
package com.fesskiev.graphql.di

import android.app.Application
import com.fesskiev.graphql.App

import javax.inject.Singleton

import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.DaggerApplication
import dagger.android.support.AndroidSupportInjectionModule

@Singleton
@Component(modules = [AndroidSupportInjectionModule::class, AppModule::class, NetworkModule::class,
    UtilsModule::class, DataSourceModule::class, ActivityBuilder::class])
interface AppComponent : AndroidInjector<DaggerApplication> {

    fun inject(app: App)

    override fun inject(instance: DaggerApplication)

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): Builder

        fun build(): AppComponent
    }
}
